### ISSUE TITLE
Maintain URL and serve proper 404 page

### ISSUE THREAD

**Issue by TimvdLippe** [sonatype-nexus-community/search-maven-org#38]
Thanks for creating an issue! Please fill out this form so we can be
sure to have all the information we need, and to minimize back and forth.

* **What are you trying to do?**
Open a page that should not exist. For example https://search.maven.org/artifact/org.mockito/mockito/2.21.0/jar should be a 404, as this artifact does not exist. Moreover, https://search.maven.org/artifact/org.mockito and https://search.maven.org/artifact/org.mockito/mockito-core/2.21.0 simply redirect to the homepage
* **What feature or behavior is this required for?**
To browse the website more nicely, users expect a 404 to happen when they entered a URL that does not exist. Moreover, the URL bar should contain the wrong the URL, such that users can fix it, without typing it all over again (for example if they made a small typo).
* **How could we solve this issue? (Not knowing is okay!)**
1. If an artifact version does not exist, show "Version <version> does not exist for artifact <artifact> of group <group>."
2. If an artifact does not exist, show "Artifact <artifact> does not exist for group <group>."
3. When only a group is provided, maybe show a list of artifacts that are in that group?
4. If a group does not exist, show "Group <group> does not exist".
* **Anything else?**

**Comment by DarthHater** [comment]
Right, we are still having some fun with 404s, given a combination of Angular SPA, nginx, and running the prior site :)

Thanks for reporting this, it is helpful (and you've been super helpful already!)

**Comment by oliviercailloux** [comment]
Also, https://search.maven.org/search?q=g:org.apache.maven.plugins%20AND%20a:maven-checkstyle-plugin&core=gav return a 404 even though the url looks correct and the page served indicates no error to the user and just serve the expected content.

Not sure whether I should open a new bug report about this or this counts as the same problem?

This behavior breaks tools that check for broken urls.
